ctapipe/io/tableloader.py

```diff
@@ -0,0 +1,308 @@
+import re
+from typing import List, Union
+
+import numpy as np
+import tables
+from astropy.table import join, vstack, Table
+
+from ..core import Component, traits
+from ..instrument import SubarrayDescription, TelescopeDescription
+from .astropy_helpers import read_table
+
+__all__ = ["get_tel_ids", "get_structure", "TableLoader"]
+
+PARAMETERS_GROUP = "/dl1/event/telescope/parameters"
+IMAGES_GROUP = "/dl1/event/telescope/images"
+GEOMETRY_GROUP = "/dl2/event/subarray/geometry"
+TRIGGER_TABLE = "/dl1/event/subarray/trigger"
+SHOWER_TABLE = "/simulation/event/subarray/shower"
+TRUE_IMAGES_GROUP = "/simulation/event/telescope/images"
+
+by_id_RE = re.compile(r"tel_\d+")
+
+
+def get_tel_ids(
+    subarray: SubarrayDescription,
+    telescopes: List[Union[int, str, TelescopeDescription]]
+) -> List[int]:
+    ids = set()
+
+    for telescope in telescopes:
+        if isinstance(telescope, int):
+            ids.add(telescope)
+        ids.update(subarray.get_tel_ids_for_type(telescope))
+
+    return sorted(ids)
+
+
+def get_structure(h5file):
+
+    if PARAMETERS_GROUP in h5file:
+        g = h5file.root[PARAMETERS_GROUP]
+    elif IMAGES_GROUP in h5file:
+        g = h5file.root[IMAGES_GROUP]
+    else:
+        raise ValueError(f"No DL1 parameters data in h5file: {h5file}")
+
+    key = next(iter(g._v_children))
+    if by_id_RE.fullmatch(key):
+        return "by_id"
+    else:
+        return "by_type"
+
+
+class TableLoader(Component):
+    """ A helper class to load and join tables from a DL1 file"""
```

Comment:
Adding a simple example here, or mentioning the main API functions that one would call would help a lot.  It took me a few tries to understand how it works.

Reply:
Since this appears in the HTML repr, it should be very concise, so maybe not a full example here (in the class docstring may be better).  Also it's not a DL1 file... but any ctapipe output file.

Reply:
done - let me know if it's enough since it cannot be very long (we have also the docs anyway - I don't think the general user care much about the class docstring...)